CMakeLists.txt

```diff
@@ -272,10 +276,32 @@ if(WIN32)
     include_directories("${CMAKE_CURRENT_SOURCE_DIR}/blosc")
 endif(WIN32)
 
+if (NOT DEFINED BLOSC_IS_SUBPROJECT)
+  if ("^${CMAKE_SOURCE_DIR}$" STREQUAL "^${PROJECT_SOURCE_DIR}$")
+    set (BLOSC_IS_SUBPROJECT FALSE)
+  else ()
+    set (BLOSC_IS_SUBPROJECT TRUE)
+    message(STATUS "Detected that BLOSC is used a subproject.")
+  endif ()
+endif ()
+
+if (NOT DEFINED BLOSC_INSTALL)
+    if (BLOSC_IS_SUBPROJECT)
+        set(BLOSC_INSTALL FALSE)
+    else()
+        set(BLOSC_INSTALL TRUE)
+    endif()
+endif()
+
 
 # subdirectories
 add_subdirectory(blosc)
 
+if (BLOSC_IS_SUBPROJECT)
+    add_library(blosc ALIAS blosc_static)
+    message(STATUS "Enabled `blosc` alias target to static library.")
```

Comment:
Why blosc needs in this case to be an alias of the static library and not of the shared one?  Just curious.